Dockerfile

```diff
@@ -38,23 +38,38 @@ FROM clearlinux:base
 LABEL maintainers="Intel"
 LABEL description="PMEM CSI Driver"
 
-# update and install needed bundles:
+# update and install needed bundles/packages:
 # file - driver uses file utility to determine filesystem type
 # xfsprogs - XFS filesystem utilities
-# storge-utils - for lvm2 and ext4(e2fsprogs) utilities
+# e2fsprogs- Ext filesystem utilities
+# LVM2-bin - for lvm2 and ext4(e2fsprogs) utilities
 ARG CACHEBUST
-RUN swupd update && swupd bundle-add file xfsprogs storage-utils && rm -rf /var/lib/swupd
+ENV PATH="/bin:/usr/bin"
+#NOTE: This below RUN command is intentionally combined all commands, to flatten the docker layers
+RUN swupd update && swupd bundle-add mixer && \
+# Create custom 'pmem-csi' bundle that holds all the needed packages
+    mixin package add file --bundle pmem-csi && \
+    mixin package add xfsprogs --bundle pmem-csi && \
+    mixin package add e2fsprogs-extras --bundle pmem-csi && \
+    mixin package add LVM2-bin --bundle pmem-csi && \
+    mixin build && \
+# Remove unneeded bundes. swupd is fail to remove child bundles, so we had to list all the bundles mixer pulled.
+# NOTE: Do not change the order of below remove bundle list, Otherwise swupd fails to remove them saying failed dependency.
+    swupd bundle-remove mixer sysadmin-basic os-installer linux-firmware-extras acpica-unix2 containers-basic git curl diffutils dnf glibc-locale gzip htop iperf iproute2 kbd kernel-install less man-pages minicom openssl parallel patch perl-basic powertop procps-ng pygobject  python3-basic strace sudo tmux tzdata unzip wpa_supplicant linux-firmware-wifi xz zstd libX11client lib-opengl lib-openssl which && \
```

Comment:
I'm worried that this long list of bundles is going to be a maintenance nightmare.

Why are these bundles installed at all? Do they get pulled in when installing mixer or are they part of the clear-linux base image?

Instead of pruning that base image, perhaps start a discussion with the Clear developers around publishing a smaller image for usage as base for containers?

Reply:
> I'm worried that this long list of bundles is going to be a maintenance nightmare.
> 
> Why are these bundles installed at all? Do they get pulled in when installing mixer or are they part of the clear-linux base image?

These are pulled by mixin, removing mixin itself is not removing all its dependencies, hence I had to remove each one of them.

Reply:
Would it be possible to run `mixin build` in one container, then start from scratch again in another where one copies over the output of that command and does the `swupd update --migrate && swupd bundle-add pmem-csi`? Then we avoid this whole fragile cleanup procedure.

If all of our containers have "swupd update" as a separate command, then the result of that command can be cached and the command only needs to run once per build.

Reply:
> Would it be possible to run `mixin build` in one container, then start from scratch again in another where one copies over the output of that command and does the `swupd update --migrate && swupd bundle-add pmem-csi`? Then we avoid this whole fragile cleanup procedure.
> 
No, it's not possible to split "mixin build" and "swupd update --migrate" to different containers, swupd loses the context to do the needed custom update.